sentence_transformers/sparse_encoder/models/MLMTransformer.py

```diff
@@ -0,0 +1,123 @@
+            config_args = {}
+
+        # Load config
+        self.config = AutoConfig.from_pretrained(model_name_or_path, cache_dir=cache_dir, **config_args)
+
+        # Load tokenizer
+        if max_seq_length is not None and "model_max_length" not in tokenizer_args:
+            tokenizer_args["model_max_length"] = max_seq_length
+
+        self.tokenizer = AutoTokenizer.from_pretrained(
+            (tokenizer_name_or_path if tokenizer_name_or_path is not None else model_name_or_path),
+            cache_dir=cache_dir,
+            **tokenizer_args,
+        )
+
+        # Set max_seq_length
+        self.max_seq_length = max_seq_length
+        if max_seq_length is None:
+            if hasattr(self.config, "max_position_embeddings") and hasattr(self.tokenizer, "model_max_length"):
+                self.max_seq_length = min(self.config.max_position_embeddings, self.tokenizer.model_max_length)
+
+        # Load MLM model
+        self.auto_model = AutoModelForMaskedLM.from_pretrained(
+            model_name_or_path, config=self.config, cache_dir=cache_dir, **model_args
+        )
+
+    def forward(self, features: dict[str, torch.Tensor]) -> dict[str, torch.Tensor]:
+        """Forward pass of the model.
+
+        Args:
+            features: Dictionary containing input features
+
+        Returns:
+            Dictionary containing token embeddings and MLM logits
+        """
+        # Get MLM outputs
+        mlm_outputs = self.auto_model(**features, output_hidden_states=True)
+
+        # Get the MLM head logits (shape: batch_size, seq_length, vocab_size)
+        mlm_logits = mlm_outputs.logits
+
+        return {"mlm_logits": mlm_logits}
+
+    def get_sentence_embedding_dimension(self) -> int:
+        """Get the dimension of the token embeddings"""
+        return self.auto_model.config.hidden_size
+
+    def tokenize(self, texts: list[str], padding: bool = True) -> dict[str, torch.Tensor]:
+        """Tokenize the input texts.
+
+        Args:
+            texts: List of texts to tokenize
+            padding: Whether to pad the sequences
+
+        Returns:
+            Dictionary containing tokenized inputs
+        """
+        # Check if the model is a DistilBERT model
+        is_distilbert = "distilbert" in self.auto_model.config.model_type.lower()
+
+        # For DistilBERT models, we need to exclude token_type_ids
+        if is_distilbert:
+            return self.tokenizer(
+                texts,
+                padding=padding,
+                truncation=True,
+                max_length=self.max_seq_length,
+                return_tensors="pt",
+                return_token_type_ids=False,  # Exclude token_type_ids for DistilBERT
+            )
+        else:
+            return self.tokenizer(
+                texts,
+                padding=padding,
+                truncation=True,
+                max_length=self.max_seq_length,
+                return_tensors="pt",
+            )
```

Comment:
Hmm, this should not be strictly necessary. The tokenizer should store the `model_input_names` which informs whether it should use the token_type_ids: https://huggingface.co/answerdotai/ModernBERT-base/blob/main/tokenizer_config.json#L939-L942

Perhaps this is indicative that a model that you tested is wrongly configured?